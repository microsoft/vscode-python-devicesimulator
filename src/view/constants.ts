// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.a

// Key events
export const CONSTANTS = {
  KEYBOARD_KEYS: {
    ENTER: "Enter"
  }
}

export default CONSTANTS;