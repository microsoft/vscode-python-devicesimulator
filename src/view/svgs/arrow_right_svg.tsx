import * as React from "react";

export const ARROW_RIGHT_SVG = (
  <svg
    width="14"
    height="14"
    viewBox="0 0 14 14"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      fill-rule="evenodd"
      clip-rule="evenodd"
      d="M8.2424 6.90907L4 11.1515L4.84853 12L9.5152 7.33333V6.48481L4.84853 1.81814L4 2.66667L8.2424 6.90907Z"
      fill="#3794FF"
    />
  </svg>
);

export default ARROW_RIGHT_SVG;
