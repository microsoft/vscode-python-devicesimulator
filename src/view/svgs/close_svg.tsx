import * as React from "react";

export const CLOSE_SVG = (
    <svg
        width="14"
        height="14"
        viewBox="0 0 14 14"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
    >
        <rect
            width="2.04536"
            height="14.7266"
            transform="matrix(0.704405 -0.709798 0.704405 0.709798 1.18579 2.45178)"
            fill="#CCCCCC"
        />
        <rect
            width="2.04536"
            height="14.7266"
            transform="matrix(0.704405 0.709798 -0.704405 0.709798 11.3735 1.09531)"
            fill="#CCCCCC"
        />
    </svg>
);

export default CLOSE_SVG;
