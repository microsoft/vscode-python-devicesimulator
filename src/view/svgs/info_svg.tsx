import * as React from "react";
import "../styles/ToolBar.css";

export const INFO_SVG = (
  <svg
    width="14"
    height="14"
    viewBox="0 0 14 14"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="info-icon-svg"
      fill-rule="evenodd"
      clip-rule="evenodd"
      d="M6.3 4.71C6.20764 4.61889 6.1351 4.50969 6.08692 4.38923C6.03873 4.26878 6.01595 4.13967 6.02 4.01C6.02 3.73 6.11 3.49 6.3 3.31C6.49 3.13 6.72 3.03 7 3.03C7.28 3.03 7.52 3.12 7.7 3.31C7.88 3.5 7.98 3.73 7.98 4.01C7.98 4.29 7.89 4.53 7.7 4.71C7.51537 4.89841 7.26377 5.00624 7 5.01C6.72 5.01 6.48 4.9 6.3 4.71ZM8 7.01C7.98 6.76 7.89 6.53 7.69 6.32C7.49 6.13 7.27 6.02 7 6.01H6C5.73 6.03 5.52 6.14 5.31 6.32C5.11 6.52 5.01 6.76 5 7.01H6V10.01C6.02 10.28 6.11 10.51 6.31 10.7C6.51 10.9 6.73 11.01 7 11.01H8C8.27 11.01 8.48 10.9 8.69 10.7C8.89 10.51 8.99 10.28 9 10.01H8V7V7.01ZM7 1.32C3.86 1.32 1.3 3.86 1.3 7C1.3 10.14 3.86 12.7 7 12.7C10.14 12.7 12.7 10.15 12.7 7C12.7 3.85 10.14 1.31 7 1.31V1.32ZM7 0C10.86 0 14 3.14 14 7C14 10.86 10.86 14 7 14C3.14 14 0 10.88 0 7C0 3.12 3.14 0 7 0Z"
    />
  </svg>
);

export default INFO_SVG;
