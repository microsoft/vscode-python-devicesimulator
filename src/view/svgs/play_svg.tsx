import * as React from "react";

export const PLAY_SVG = (
  <svg
    width="14"
    height="14"
    viewBox="0 0 14 14"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="play-icon"
      d="M1.00005 0.854163C1.00003 0.471921 1.41151 0.231022 1.74482 0.418142L12.487 6.44883C12.8243 6.63823 12.828 7.12258 12.4936 7.31706L1.75209 13.5631C1.41877 13.757 1.00077 13.5165 1.00075 13.1309L1.00005 0.854163Z"
      fill="#23D18B"
    />
  </svg>
);

export default PLAY_SVG;
