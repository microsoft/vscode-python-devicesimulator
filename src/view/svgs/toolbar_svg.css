.svg-path {
  fill: var(--vscode-badge-foreground);
}
