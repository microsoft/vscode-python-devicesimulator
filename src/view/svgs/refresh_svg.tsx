import * as React from "react";
import "./toolbar_svg.css";

export const REFRESH_SVG = (
  <svg
    width="14"
    height="14"
    viewBox="0 0 14 14"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      fill-rule="evenodd"
      clip-rule="evenodd"
      d="M7.12508 4.13728V2.92015C4.59083 2.92015 2.53127 4.74924 2.53127 6.9999C2.53127 7.53706 2.64611 8.06063 2.86815 8.5298C2.98299 8.77458 2.89877 9.05336 2.69205 9.23695C2.30158 9.58373 1.64313 9.46134 1.43641 9.00577C1.15313 8.38701 1 7.70705 1 6.9999C1 3.99449 3.74097 1.56024 7.12508 1.56024V0.343119C7.12508 0.0371382 7.53852 -0.112452 7.77587 0.0983343L9.91199 1.99541C10.0651 2.1314 10.0651 2.34219 9.91199 2.47818L7.77587 4.37526C7.53852 4.59285 7.12508 4.44326 7.12508 4.13728ZM11.3818 5.46997C11.267 5.23199 11.3512 4.94641 11.5579 4.76282C11.9484 4.41604 12.6068 4.53843 12.8136 4.994C13.0969 5.61277 13.25 6.29272 13.25 6.99988C13.25 10.0053 10.509 12.4395 7.1249 12.4395V13.6635C7.1249 13.9626 6.71146 14.1122 6.47411 13.9014L4.33799 12.0044C4.18487 11.8684 4.18487 11.6576 4.33799 11.5216L6.47411 9.62451C6.71146 9.40692 7.1249 9.55651 7.1249 9.8625V11.0796C9.65915 11.0796 11.7187 9.25053 11.7187 6.99988C11.7187 6.46271 11.5962 5.94594 11.3818 5.46997Z"
      className="svg-path"
    />
  </svg>
);

export default REFRESH_SVG;
