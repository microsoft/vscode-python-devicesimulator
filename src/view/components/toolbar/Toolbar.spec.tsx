import * as React from "react";
import * as ReactDOM from "react-dom";
import { IntlProvider } from "react-intl";
import * as testRenderer from "react-test-renderer";
import Toolbar from "./ToolBar";
import * as TOOLBAR_SVG from "../../svgs/toolbar_svg";
import { MICROBIT_TOOLBAR_ID } from "./SensorModalUtils";
import { SENSOR_LIST } from "../../constants";

const MOCK_TOOLBAR_BUTTONS: Array<{ label: string; image: JSX.Element }> = [
    {
        image: TOOLBAR_SVG.LIGHT_SVG,
        label: MICROBIT_TOOLBAR_ID.LIGHT,
    },
    {
        image: TOOLBAR_SVG.MOTION_SVG,
        label: MICROBIT_TOOLBAR_ID.ACCELEROMETER,
    },
];
const mockUpdateSensors = () => {
    return;
};
const mockInitialValues = {
    [SENSOR_LIST.TEMPERATURE]: 0,
    [SENSOR_LIST.LIGHT]: 0,
};
describe("Toolbar component ", () => {
    it("should render correctly", () => {
        const component = testRenderer
            .create(
                <IntlProvider locale="en">
                    <Toolbar
                        buttonList={MOCK_TOOLBAR_BUTTONS}
                        onUpdateSensor={mockUpdateSensors}
                        sensorValues={mockInitialValues}
                    />
                </IntlProvider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <IntlProvider locale="en">
                <Toolbar
                    buttonList={MOCK_TOOLBAR_BUTTONS}
                    onUpdateSensor={mockUpdateSensors}
                    sensorValues={mockInitialValues}
                />
            </IntlProvider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});
