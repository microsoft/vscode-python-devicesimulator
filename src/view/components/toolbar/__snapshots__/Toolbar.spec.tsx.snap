// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component  should render correctly 1`] = `
<div
  className="toolbar-parent"
  id="toolbar-parent"
>
  <div
    className="toolbar"
  >
    <div
      className="toolbar-icon"
    >
      <div
        className="ms-TooltipHost root-49"
        onBlurCapture={[Function]}
        onFocusCapture={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          aria-label="toolbar light-sensor"
          className="toolbar-button button"
          id="toolbar-light-sensor-button"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "width": 32,
            }
          }
          tabIndex={0}
        >
          <svg
            fill="none"
            height="16px"
            version="1.1"
            viewBox="0 0 16 16"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <title>
              Light sensor
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
              id="light_sensor"
              stroke="none"
              stroke-width="1"
            >
              <g
                className="button-icon"
                fill-rule="nonzero"
                id="Group-10"
                transform="translate(4.000000, 0.000000)"
              >
                <g
                  id="light-bulb"
                >
                  <path
                    d="M8.30374549,6.20872934 L7.44295783,8.17745143 C7.23654761,8.7386575 7.20852246,8.7656952 6.99231195,9.20813126 L6.45540049,10.4361106 C6.46864532,10.5933327 6.46346172,11 6.05537849,11 L2.94465012,11 C2.68129241,11 2.58708458,10.8227848 2.55655729,10.6570975 L2.54351121,10.6570975 L0,4.83978978 L0.0886634694,4.84519734 C0.0290289823,4.57964193 -0.00106181819,4.307929 2.86084252e-05,4.03423261 C0.00757892,1.8945771 1.94423384,0.0877568873 4.31880683,0.00284992256 C5.55705792,-0.0379054205 6.72735622,0.359459175 7.61829298,1.13041441 C8.50922975,1.89797338 9,2.93723463 9,4.04781772 C9,4.82474009 8.75968094,5.56322682 8.30374549,6.20872934 Z"
                    id="Shape"
                  />
                  <path
                    d="M6.5,12.4 C6.82926694,12.4 7.1,12.6707331 7.1,13 C7.1,13.3307854 6.83256829,13.6 6.5,13.6 L2.5,13.6 C2.16921459,13.6 1.9,13.3325683 1.9,13 C1.9,12.6692146 2.16743171,12.4 2.5,12.4 L6.5,12.4 Z"
                    id="Shape"
                  />
                  <path
                    d="M5.66965517,14.8 C5.96435528,14.8 6.2,15.0737859 6.2,15.4 C6.2,15.7277609 5.9672384,16 5.66965517,16 L3.53034483,16 C3.23437963,16 3,15.7294873 3,15.4 C3,15.0722391 3.2327616,14.8 3.53034483,14.8 L5.66965517,14.8 Z"
                    id="Shape"
                  />
                </g>
              </g>
            </g>
          </svg>
        </button>
      </div>
      <div
        className="ms-TooltipHost root-49"
        onBlurCapture={[Function]}
        onFocusCapture={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          aria-label="toolbar accelerometer-sensor"
          className="toolbar-button button"
          id="toolbar-accelerometer-sensor-button"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "width": 32,
            }
          }
          tabIndex={0}
        >
          <svg
            fill="none"
            height="16px"
            version="1.1"
            viewBox="0 0 17 16"
            width="17px"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              id="motion_sensor"
              stroke="none"
              stroke-width="1"
            >
              <g
                className="button-icon"
                id="Group-10"
              >
                <g
                  id="Group-12"
                >
                  <path
                    d="M8,8.8203 L5.88,11.9103 L8,11.9103 L8,12.5493 L5,12.5493 L5,12.0303 L7.12,8.9403 L5.12,8.9403 L5.12,8.2993 L8,8.2993 L8,8.8203 Z M17,10.5903 L13.87,8.7903 L13.87,9.9903 L11.76,9.9903 C11.63,8.7803 11.1,7.6503 10.25,6.7693 C9.39,5.9103 8.25,5.3703 7.04,5.2403 L7.04,3.1203 L8.23,3.1203 L6.38,0.0003 L4.63,3.1203 L5.83,3.1203 L5.83,5.2403 C4.5,5.3803 3.27,6.0003 2.38,6.9803 C1.48,7.9693 0.99,9.2503 1,10.5703 C0.99,11.2803 1.12,11.9903 1.38,12.6393 C1.65,13.2993 2.04,13.9003 2.54,14.4103 C3.04,14.9103 3.64,15.3093 4.3,15.5903 C4.96,15.8603 5.66,16.0003 6.38,16.0003 C7.72,16.0093 9.01,15.5193 10.01,14.6303 C11,13.7403 11.63,12.5093 11.76,11.1903 L13.87,11.1903 L13.87,12.3893 L17,10.5903 Z"
                    id="Fill-1"
                  />
                  <polygon
                    id="Fill-3"
                    points="2.3398 4 2.9998 4 2.9998 2.42 4.3398 0 3.6108 0 2.6698 1.75 1.7598 0 0.9998 0 2.3398 2.4"
                  />
                  <polygon
                    id="Fill-10"
                    points="14.54 4.0098 13.69 5.5198 12.84 4.0098 12.05 4.0098 13.269 6.0098 12 8.1398 12.769 8.1398 13.67 6.5498 14.58 8.1398 15.37 8.1398 14.08 5.9998 15.31 3.9998"
                  />
                </g>
              </g>
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
